Fix sphinx automation

myeatman-bdai opened the pull request:
The workflow to generate sphinx documentation seems to not be working. Trying to fix that with this PR.

myeatman-bdai commented:
@petercorke , it appears that the sphinx documentation workflow was dependent on a repo you had forked, `sphinx-autorun` (if I knew this I had forgotten). Some changes you made 3 weeks ago had a knock-on effect here. I've pinned the git hash in the automation to a previous version.

petercorke commented:
I made some mods to speed it up, and change it to use project.toml, but that file is broken.  Will look at that next

petercorke commented on README.md:
```diff
@@ -20,7 +20,7 @@
 <img src="https://github.com/bdaiinstitute/spatialmath-python/raw/master/docs/figs/CartesianSnakes_LogoW.png" width="200"></td>
 <td style="border:0px">
-A Python implementation of the <a href="https://github.com/petercorke/spatial-math">Spatial Math Toolbox for MATLAB<sup>&reg;</sup></a>
+A Python implementation of the <a href="https://github.com/bdaiinstitute/spatial-math">Spatial Math Toolbox for MATLAB<sup>&reg;</sup></a>
```
This change is wrong, it refers to the MATLAB ancestor which is surely not at BDAI.

myeatman-bdai replied:
Ah, sorry about that!

petercorke commented on README.md:
```diff
@@ -80,7 +80,7 @@ The class, method and functions names largely mirror those of the MATLAB toolbox
 # Citing
 
-Check out our ICRA 2021 paper on [IEEE Xplore](https://ieeexplore.ieee.org/document/9561366) or get the PDF from [Peter's website](https://bit.ly/icra_rtb).  This describes the [Robotics Toolbox for Python](https://github.com/petercorke/robotics-toolbox-python) as well Spatial Maths.
+Check out our ICRA 2021 paper on [IEEE Xplore](https://ieeexplore.ieee.org/document/9561366) or get the PDF from [Peter's website](https://bit.ly/icra_rtb).  This describes the [Robotics Toolbox for Python](https://github.com/bdaiinstitute/robotics-toolbox-python) as well Spatial Maths.
```
last line, missing word "as well **as**"

myeatman-bdai replied:
:+1:

petercorke commented on README.md:
```diff
@@ -447,6 +447,6 @@ Similarly when we assign an element or slice of the symbolic matrix to a numeric
 ## History & Contributors
 
-This package was originally created by [Peter Corke](https://github.com/petercorke) and [Jesse Haviland](https://github.com/jhavl) and was inspired by the [Spatial Math Toolbox for MATLAB](https://github.com/petercorke/spatialmath-matlab).  It supports the textbook [Robotics, Vision & Control in Python 3e](https://github.com/petercorke/RVC3-python).
+This package was originally created by [Peter Corke](https://github.com/bdaiinstitute) and [Jesse Haviland](https://github.com/jhavl) and was inspired by the [Spatial Math Toolbox for MATLAB](https://github.com/bdaiinstitute/spatialmath-matlab).  It supports the textbook [Robotics, Vision & Control in Python 3e](https://github.com/bdaiinstitute/RVC3-python).
```
Please undo the last 2 changes in this block.

myeatman-bdai replied:
:+1:

petercorke commented on docs/source/intro.rst:
```diff
@@ -718,9 +718,9 @@ Relationship to MATLAB tools
 
 This package replicates, as much as possible, the functionality of the `Spatial
-Math Toolbox  <https://github.com/petercorke/spatial-math>`__ for MATLAB®
+Math Toolbox  <https://github.com/bdaiinstitute/spatial-math>`__ for MATLAB®
```
revert

myeatman-bdai replied:
:+1:

petercorke commented on docs/source/intro.rst:
```diff
@@ -718,9 +718,9 @@ Relationship to MATLAB tools
+Math Toolbox  <https://github.com/bdaiinstitute/spatial-math>`__ for MATLAB®
 which underpins the `Robotics Toolbox
-<https://github.com/petercorke/robotics-toolbox-matlab>`__ for MATLAB®. It
+<https://github.com/bdaiinstitute/robotics-toolbox-matlab>`__ for MATLAB®. It
```
revert

myeatman-bdai replied:
:+1:

petercorke commented on README.md:
```diff
@@ -1,6 +1,6 @@
 # Spatial Maths for Python
 
-[![A Python Robotics Package](https://raw.githubusercontent.com/petercorke/robotics-toolbox-python/master/.github/svg/py_collection.min.svg)](https://github.com/petercorke/robotics-toolbox-python)
+[![A Python Robotics Package](https://raw.githubusercontent.com/petercorke/robotics-toolbox-python/master/.github/svg/py_collection.min.svg)](https://github.com/bdaiinstitute/robotics-toolbox-python)
```
revert, this toolbox is not a BDAI one (but we can talk about that :)

myeatman-bdai replied:
:+1:

petercorke reviewed (changes requested):
I believe I have fixed the `sphinx-auto` problem, bad `pyproject.toml` file.  It is probably a 100x faster now, but just go with what's most expedient for you.  It has revealed quite a few execution errors in the example code blocks (the example blocks are wrong, not the toolbox), I will make a PR for these, but this shouldn't stop us publishing.

There are a few incorrect changes from `petercorke` to `bdaiinstitute` when referencing robotics toolbox, RVC book, MATLAB ancestor of this toolbox.

A lot of changes seem to be just trailing white space.

myeatman-bdai commented:
> I believe I have fixed the `sphinx-auto` problem, bad `pyproject.toml` file. It is probably a 100x faster now, but just go with what's most expedient for you. It has revealed quite a few execution errors in the example code blocks (the example blocks are wrong, not the toolbox), I will make a PR for these, but this shouldn't stop us publishing.
> 
> There are a few incorrect changes from `petercorke` to `bdaiinstitute` when referencing robotics toolbox, RVC book, MATLAB ancestor of this toolbox.
> 
> A lot of changes seem to be just trailing white space.

Sorry about that, I did a blanket replace of "github/petercorke" as a stab at figuring out what was going on and thought I had reverted it already.

myeatman-bdai commented:
> A lot of changes seem to be just trailing white space.

I thought the black formatting PR I just merged would have also fixed this for all the files as well. I'll probably make one for that then.

petercorke reviewed (approved):
This looks great, thanks
